include/swift/AST/DiagnosticsFrontend.def

```diff
@@ -209,6 +209,10 @@ ERROR(symbol_in_ir_not_in_tbd,none,
       "symbol '%0' (%1) is in generated IR file, but not in TBD file",
       (StringRef, StringRef))
 
+ERROR(redundant_prefix_compilation_flag,none,
+      "did you provide a redundant '-D' in your build settings? (got '-D%0')", 
```

Comment:
We usually try to say what went wrong first and then offer a suggestion, so maybe something more like "invalid argument '-D%0'; did you…"?

Reply:
Thank you! I will try to fix message!! ✨